time/date.jsonnet

```diff
@@ -34,8 +34,52 @@ local months_seconds(year, month) = std.foldl(
 // days_seconds returns the number of seconds in all days up to day-1.
 local days_seconds(day) = (day - 1) * 24 * 3600;
 
+// is_numeric returns a bool indicating whether the input string is numeric
+local is_numeric(input) = std.length(input) > 0 && std.length(std.prune([
+  if std.codepoint('0') > std.codepoint(char) || std.codepoint(char) > std.codepoint('9') then char
+  for char in std.stringChars(input)
+])) == 0;
```

Comment:
I don't think we tend to be particularly defensive with our functions but I would consider adding a type assertion for string input.

```jsonnet
local is_numeric(input) =
  local type = std.type(input);
  assert type == 'string' : 'is_numeric() only operates on strings, got %s' % type;
  EXPR;
```

For the per codepoint check, I think it would be more idiomatic in functional programming to use a fold function. The following `foldl` starts with an accumulator, `acc`, set to the boolean result of `std.length(input) > 0` and folds the result of the test for digits with that accumulator. I've inverted the test from your code because I personally find it easier to think in this way but you can flip it back if you like.

```jsonnet
std.foldl(
  function(acc, char) acc && std.codepoint('0') <= std.codepoint(char) && std.codepoint(char) <= std.codepoint('9'),
  std.stringChars(input),
  std.length(input) > 0,
)
```

Reply:
Oh right... functional programming. I guess I have to switch my mindset as I'm still doing imperative in jsonnet.